google-cloud-redis/src/test/java/com/google/cloud/redis/v1beta1/it/ITSystemTest.java

```diff
@@ -0,0 +1,111 @@
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.redis.v1beta1.CloudRedisClient;
+import com.google.cloud.redis.v1beta1.Instance;
+import com.google.cloud.redis.v1beta1.InstanceName;
+import com.google.cloud.redis.v1beta1.LocationName;
+import com.google.cloud.redis.v1beta1.UpdateInstanceRequest;
+import com.google.common.collect.Lists;
+import com.google.protobuf.FieldMask;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ITSystemTest {
+
+  private static CloudRedisClient client;
+
+  private static final Logger LOG = Logger.getLogger(ITSystemTest.class.getName());
+  private static final String PROJECT_ID = ServiceOptions.getDefaultProjectId();
+  private static final String NETWORK = System.getProperty("redis.network", "default");
+  private static final String INSTANCE =
+      "test-instance-" + UUID.randomUUID().toString().substring(0, 8);
+  private static final String LOCATION = "us-central1";
+  private static final String AUTHORIZED_NETWORK =
+      "projects/" + PROJECT_ID + "/global/networks/" + NETWORK;
+
+  private static final Instance.Tier TIER = Instance.Tier.BASIC;
+  private static final LocationName PARENT = LocationName.of(PROJECT_ID, LOCATION);
+  private static final InstanceName INSTANCE_NAME = InstanceName.of(PROJECT_ID, LOCATION, INSTANCE);
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    client = CloudRedisClient.create();
+    /* Creates a Redis instance based on the specified tier and memory size. */
+    Instance instance =
+        Instance.newBuilder()
+            .setTier(TIER)
+            .setMemorySizeGb(1)
+            .setAuthorizedNetwork(AUTHORIZED_NETWORK)
+            .build();
+    client.createInstanceAsync(PARENT, INSTANCE, instance).get();
+    LOG.info("redis instance created successfully.");
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    /* Deletes a specific Redis instance. Instance stops serving and data is deleted. */
+    client.deleteInstanceAsync(INSTANCE_NAME);
+    LOG.info("redis instance deleted successfully.");
+    client.close();
+  }
+
+  @Test
+  public void testGetInstance() {
+    Instance response = client.getInstance(INSTANCE_NAME);
+    assertEquals(TIER, response.getTier());
+    assertEquals(INSTANCE_NAME.toString(), response.getName());
+  }
+
+  @Test
+  public void testListInstances() {
+    CloudRedisClient.ListInstancesPagedResponse pagedListResponse = client.listInstances(PARENT);
+    List<Instance> resources = Lists.newArrayList(pagedListResponse.iterateAll());
+    int instance = 0, count = 0;
+    while (instance < resources.size()) {
+      count++;
+      instance++;
+    }
+    assertEquals(count, resources.size());
+  }
```

Comment:
```
    int instance = 0, count = 0;
    while (instance < resources.size()) {
      count++;
      instance++;
    }

```
This code doesn't check anything. It will never fail.
It's better to either remove this test at all, or replace this code with 
assertEquals(X, resources.size()) , where X - some constant, expected value for resource size.

Reply:
done

Reply:
good!